### PR TITLE
Expand on the `<group>` element

### PR DESCRIPTION
**Pull request by javimed**
## Description
This PR clarifies the rule <group> tags.

### Docs building
- [x] Compiles without warnings.
### Code formatting and web optimization
- [x] Uses three spaces indentation.
- [x] Adds or updates meta descriptions accordingly.
- [x] Updates the `redirects.js` script if necessary (check [this guide](https://github.com/wazuh/wazuh-documentation/blob/master/NEW_RELEASE.md)).
### Writing style
- [x] Uses present tense, active voice, and semi-formal registry.
- [x] Uses short, simple sentences.
- [x] Uses **bold** for user interface elements, _italics_ for key terms or emphasis, and `code` font for Bash commands, file names, REST paths, and code.

### CODE REVIEW

**Comment by JorgeMarinoDev** [comment]
The main lack of documentation here was related to an ambiguity of the token in the rules definitions.

can be used enclosing a whole rule or inside the rule itself.

Both cases have totally different purposes and source/user-manual/ruleset/ruleset-xml-syntax/rules.rst was updated accordingly.

**Comment by cborla** [comment]
@javimed, I will do the first review of the PR changes, but we need you to do the final review and approve it if you see everything right, as you have generated the issue you can't do the review, but leave a comment approving the PR if you see everything right.

File: source/user-manual/ruleset/ruleset-xml-syntax/rules.rst
Comment (JorgeMarinoDev) [reply source/user-manual/ruleset/ruleset-xml-syntax/rules.rst:220]: @javimed This is something we have already discussed. After lab tests it was confirmed that enclosing <group> is only for reporting purposes. If another rule needs to filter with <if_group> it did not work. Maybe it's an expected behavior. As far as it goes, Documentation shall be adapted to the current behavior. And we are aligned with this. Thank you. Already reported this to the main development team's channel. Good job .

Comment (javimed) [reply source/user-manual/ruleset/ruleset-xml-syntax/rules.rst:220]: Then it's a LGTM

Comment (javimed) [reply source/user-manual/ruleset/ruleset-xml-syntax/rules.rst:220]: I ran a test and it worked fine in this case

Comment (javimed) [inline source/user-manual/ruleset/ruleset-xml-syntax/rules.rst:220]:
```suggestion
To define rules that trigger only if another rule in a specific group has triggered, check the `if_group`_ and `if_matched_group`_ options.
```

Comment (javimed) [reply source/user-manual/ruleset/ruleset-xml-syntax/rules.rst]: Removed the sentence because I ran a test and it worked fine.

Comment (cborla) [reply source/user-manual/ruleset/ruleset-xml-syntax/rules.rst]: LGTM!

**Review by cborla — APPROVED** [review]
LGTM!

**Pull request merged**
